File: FindTheDifference/fed.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        dic = {}
        for n in s:
            if n in dic.keys():dic[n] += 1
            else: dic[n] = 1
        for n in t:
            if n in dic.keys() and dic[n]>0:dic[n]-=1
            else: return n
        return 

class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        arr1 = [0]*26
        for n in s: 
            arr1[ord(n)-ord('a')]+=1
        for n in t:
            if arr1[ord(n)-ord('a')]>0:
                arr1[ord(n)-ord('a')]-=1
            else :return n
        return 

File: FindTheDifference/fed.java
class Solution {
    public char findTheDifference(String s, String t) {
        HashMap<Character,Integer> map = new HashMap<Character,Integer>();
        for(int i=0;i<s.length();i++){
            char c = s.charAt(i);
            if(map.containsKey(c)) map.put(c,map.get(c)+1);
            else map.put(c,1);
        }
        for(int i=0;i<t.length();i++){
            char c = t.charAt(i);
            if(map.containsKey(c) && map.get(c)>0) map.put(c,map.get(c)-1);
            else return c;
        }
        return 'a';
    }
}

class Solution {
    public char findTheDifference(String s, String t) {
        int[] arr = new int[26];
        for(int i = 0;i<26;i++){
            arr[i] = 0;
        }
        for(int i =0;i<s.length();i++){
            char c = s.charAt(i);
            arr[c-'a']++;
        }
        for(int i=0;i<t.length();i++){
            char c = t.charAt(i);
            if(arr[c-'a']<1) return c;
            arr[c-'a']--;
        }
        return 'a';
    }
}